test_cases/stdlib/builtins/check_reversed.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from typing import Generic, TypeVar
+from typing_extensions import assert_type
+
+x: list[int] = []
+assert_type(list(reversed(x)), "list[int]")
+
+
+class MyReversible:
+    def __iter__(self) -> Iterator[str]:
+        yield "blah"
```

Comment:
```suggestion
    def __iter__(self) -> Iterator[int]:
        yield 1
```
Use a different type, so we check that the type of `__reversed__` is picked up